addons/binding/org.openhab.binding.plclogo/src/main/java/org/openhab/binding/plclogo/internal/discovery/PLCDiscoveryService.java

```diff
@@ -0,0 +1,164 @@
+/**
+ * Copyright (c) 2010-2018 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.plclogo.internal.discovery;
+
+import static org.openhab.binding.plclogo.PLCLogoBindingConstants.THING_TYPE_DEVICE;
+
+import java.io.IOException;
+import java.net.Inet4Address;
+import java.net.InetAddress;
+import java.net.InterfaceAddress;
+import java.net.NetworkInterface;
+import java.net.SocketException;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Enumeration;
+import java.util.Objects;
+import java.util.Set;
+import java.util.TreeSet;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.RejectedExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.locks.ReentrantLock;
+
+import org.apache.commons.net.util.SubnetUtils;
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.config.discovery.AbstractDiscoveryService;
+import org.eclipse.smarthome.config.discovery.DiscoveryResultBuilder;
+import org.eclipse.smarthome.config.discovery.DiscoveryService;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.thing.ThingUID;
+import org.eclipse.smarthome.model.script.actions.Ping;
+import org.osgi.service.component.annotations.Component;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link PLCDiscoveryService} is responsible for discovering devices on
+ * the current Network. It uses every Network Interface which is connected to a network.
+ * Based on network binding discovery service.
+ *
+ * @author Alexander Falkenstern - Initial contribution
+ */
+@NonNullByDefault
+@Component(service = DiscoveryService.class)
+public class PLCDiscoveryService extends AbstractDiscoveryService {
+
+    private final Logger logger = LoggerFactory.getLogger(PLCDiscoveryService.class);
+
+    // Bridge config properties
+    private static final String LOGO_HOST = "address";
+    private static final int LOGO_PORT = 102;
+
+    private static final int DISCOVERY_TIMEOUT = 30;
+    private static final Set<ThingTypeUID> THING_TYPES_UIDS = Collections.singleton(THING_TYPE_DEVICE);
+
+    private static final int CONNECTION_TIMEOUT = 500;
+    private Set<String> addresses = new TreeSet<>();
+
+    private @Nullable ExecutorService executor;
```

Comment:
Does it make sense to keep the thread pool? It might be better, to shut it down after use and create a new one upon the next scan (as they are manually triggered, they do not happen too often).